src/network/components/StreamSearchDropdown.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { FunctionComponent, useEffect, useState } from 'react'
+import { useQuery } from '@tanstack/react-query'
+import { truncateStreamName } from '~/shared/utils/text'
+import { SearchDropdown } from '~/components/SearchDropdown'
+import {
+    checkIfStreamExists,
+    getPagedStreams,
+    TheGraphOrderBy,
+    TheGraphOrderDirection,
+    TheGraphStreamResult,
+} from '~/services/streams'
+
+export const StreamSearchDropdown: FunctionComponent<{
+    onStreamChange: (streamId: string) => void
+    name: string
+    streamId: string
+    disabled?: boolean
+}> = ({ onStreamChange, name, streamId, disabled }) => {
+    const [streamSearchValue, setStreamSearchValue] = useState('')
+
+    const streamsQuery = useQuery({
+        queryKey: ['createSponsorshipsStreamSearch', streamSearchValue],
+        queryFn: async (ctx) => {
+            const result: TheGraphStreamResult = await getPagedStreams(
+                20,
+                ctx.pageParam,
+                undefined,
+                streamSearchValue,
+                TheGraphOrderBy.Id,
+                TheGraphOrderDirection.Asc,
+            )
+
+            return result
+        },
+        staleTime: 60 * 1000, // 1 minute
+        keepPreviousData: true,
+    })
+
+    useEffect(() => {
+        if (!streamId || streamId === streamSearchValue) {
+            return
+        }
+        checkIfStreamExists(streamId).then((exists) => {
+            console.log('exists?', exists)
+            if (exists) {
+                onStreamChange(streamId)
+                setStreamSearchValue(streamId)
+            }
+        })
+        // eslint-disable-next-line
```

Comment:
Can you please add a comment to it? I bet in a week I'mma be scratching my head thinking whats and whys.

```suggestion
        // eslint-disable-next-line react-hooks/exhaustive-deps
```